core/src/OpenACC/Kokkos_OpenACC_Team.hpp

```diff
@@ -0,0 +1,616 @@
+      : m_team_scratch_size{0, 0},
+        m_thread_scratch_size{0, 0},
+        m_tune_team_size(false),
+        m_tune_vector_length(true),
+        m_chunk_size(0) {
+    init(league_size_request, team_size_request, 1);
+  }
+  static size_t vector_length_max() {
+    return 32; /* TODO: this is bad. Need logic that is compiler and backend
+                  aware */
+  }
+  int team_alloc() const { return m_team_alloc; }
+  int team_iter() const { return m_team_iter; }
+
+  int chunk_size() const { return m_chunk_size; }
+
+  /** \brief set chunk_size to a discrete value*/
+  TeamPolicyInternal& set_chunk_size(typename traits::index_type chunk_size_) {
+    m_chunk_size = chunk_size_;
+    return *this;
+  }
+
+  /** \brief set per team scratch size for a specific level of the scratch
+   * hierarchy */
+  TeamPolicyInternal& set_scratch_size(const int& level,
+                                       const PerTeamValue& per_team) {
+    m_team_scratch_size[level] = per_team.value;
+    return *this;
+  }
+
+  /** \brief set per thread scratch size for a specific level of the scratch
+   * hierarchy */
+  TeamPolicyInternal& set_scratch_size(const int& level,
+                                       const PerThreadValue& per_thread) {
+    m_thread_scratch_size[level] = per_thread.value;
+    return *this;
+  }
+
+  /** \brief set per thread and per team scratch size for a specific level of
+   * the scratch hierarchy */
+  TeamPolicyInternal& set_scratch_size(const int& level,
+                                       const PerTeamValue& per_team,
+                                       const PerThreadValue& per_thread) {
+    m_team_scratch_size[level]   = per_team.value;
+    m_thread_scratch_size[level] = per_thread.value;
+    return *this;
+  }
+
+ private:
+  /** \brief finalize chunk_size if it was set to AUTO*/
+  void set_auto_chunk_size() {
+    int concurrency = 2048 * default_team_size;
+
+    if (m_chunk_size > 0) {
+      if (!Impl::is_integral_power_of_two(m_chunk_size))
+        Kokkos::abort("TeamPolicy blocking granularity must be power of two");
+    }
+
+    int new_chunk_size = 1;
+    while (new_chunk_size * 100 * concurrency < m_league_size)
+      new_chunk_size *= 2;
+    if (new_chunk_size < default_team_size) {
+      new_chunk_size = 1;
+      while ((new_chunk_size * 40 * concurrency < m_league_size) &&
+             (new_chunk_size < default_team_size))
+        new_chunk_size *= 2;
+    }
+    m_chunk_size = new_chunk_size;
+  }
+
+ public:
+  using member_type = Impl::OpenACCTeamMember;
+};
+}  // namespace Impl
+}  // namespace Kokkos
+
+namespace Kokkos {
+namespace Impl {
+
+template <typename iType>
+struct TeamThreadRangeBoundariesStruct<iType, OpenACCTeamMember> {
+  using index_type = iType;
+  const iType start;
+  const iType end;
+  const OpenACCTeamMember& team;
+
+  inline TeamThreadRangeBoundariesStruct(const OpenACCTeamMember& thread_,
```

Comment:
```suggestion
  TeamThreadRangeBoundariesStruct(const OpenACCTeamMember& thread_,
```
Same comment for all the constructors below

Reply:
Done.